### PR TITLE
Remove singleton usage of classes wherever possible

### PR DESCRIPTION
**Pull request by loehnertz**

### CODE REVIEW

File: src/main/kotlin/service/user/balance/BalanceIncreaseRateManager.kt [hunk 1]
```diff
@@ -17,28 +17,16 @@ class BalanceIncreaseRateManager(val user: User) {
...
     }
 
     fun retrieveIncreaseRate(): Long? {
-        val redis = RedisFactory.retrieveRedisClient()
-
         return try {
-            redis.hmget(RedisHashMapKeyIncreaseRates, user.name).firstOrNull()?.toLong()
+            RedisConnector().hmget(RedisHashMapKeyIncreaseRates, user.name).firstOrNull()?.toLong()
         } catch (exception: Exception) {
             // TODO: Add logging here
```
Comment (SimonBaars) [inline src/main/kotlin/service/user/balance/BalanceIncreaseRateManager.kt:28]: Are you keeping track of the TODO's?

File: src/main/kotlin/service/user/balance/BalanceManager.kt [hunk 2]
```diff
@@ -4,27 +4,19 @@ import model.User
...
 
     fun retrieveCurrentBalance(): Long {
-        val redis = RedisFactory.retrieveRedisClient()
-
-        val currentBalance = redis.hmget(RedisKeyUserBalances, user.name).firstOrNull()?.toLong()
-
-        redis.close()
+        val currentBalance = RedisConnector().hmget(RedisKeyUserBalances, user.name).firstOrNull()?.toLong()
 
         return if (currentBalance != null) {
```
Comment (SimonBaars) [inline src/main/kotlin/service/user/balance/BalanceManager.kt:21]: Interesting syntax this `return if`. Does it return null instead if the condition is not met?

**Review by SimonBaars — APPROVED** [review]
:sun_with_face:

File: src/main/kotlin/service/user/balance/BalanceManager.kt (hunk 2 above)
Comment (loehnertz) [reply src/main/kotlin/service/user/balance/BalanceManager.kt:21]: No, it returns the last statement in the `else` block.

File: src/main/kotlin/service/user/balance/BalanceIncreaseRateManager.kt (hunk 1 above)
Comment (loehnertz) [reply src/main/kotlin/service/user/balance/BalanceIncreaseRateManager.kt:28]: Yep, will be in the next PR!

File: src/main/kotlin/service/user/balance/BalanceManager.kt (hunk 2 above)
Comment (SimonBaars) [reply src/main/kotlin/service/user/balance/BalanceManager.kt:21]: Oh cool! I guess in kotlin an if statement is an expression, meaning you can put it in lots of places (like with Haskell).

**Review by CJStuart — APPROVED** [review]

**Pull request merged**